Microsoft.AVS.VMFS/Microsoft.AVS.VMFS.psm1

```diff
@@ -568,3 +568,262 @@ function Remove-VMHostStaticIScsiTargets {
+     ESXi host IP Address 
+
+    .PARAMETER HostAdapter
+     ESXi host storage adapter name 
+
+    .PARAMETER NodeAddress
+     Storage Node EndPoint Address
+
+    .PARAMETER StorageSystemNQN
+     Storage system NQN
+    
+    .PARAMETER  AdminQueueSize
+     NVMe/TCP Admin Queue Size, default 32
+
+    .PARAMETER  ControllerId
+     NVMe/TCP Controller ID, default 65535 
+
+    .PARAMETER  IoQueueNumber
+     IO Queue Number, default 8
+
+    .PARAMETER IoQueueSize
+     IO Queue Size, default 256
+     
+    .PARAMETER KeepAliveTimeout
+     Keep Alive Timeout, default 256
+     
+    .PARAMETER  PortNumber
+     Target Port Number, default 4420
+
+    .EXAMPLE
+     Connect-NVMeTCPTarget HostAddress "192.168.0.1" -HostAdapter "adapter-name" -NodeAddress "192.168.0.1" -StorageSystemNQN "nqn.2016-01.com.lightbitslabs:uuid:46edb489-ba18-4dd4-a157-1d8eb8c32e21"
+
+    .INPUTS
+     ESXi Address, Storage Adapter, Storage Node Address, Storage System NQN
+
+    .OUTPUTS
+     None.
+#>
+function Connect-NVMeTCPTarget {
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+    Param
+    (
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'ESXi host network address')]
+        [string] $HostAddress,
+
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'NVMe/TCP Storage Adapter Name')]
+        [string] $HostAdapter,
+
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'Target storage Node datapath address')]
+        [string]     $NodeAddress,
+
+
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'Target storage SystemNQN')]
+        [string]     $StorageSystemNQN,
+
+        [Parameter(
+            Mandatory = $false,
+            HelpMessage = 'NVMe/TCP Admin Queue Size')]
+        [int]     $AdminQueueSize = 32,
+
+        [Parameter(
+            Mandatory = $false,
+            HelpMessage = 'NVMe/TCP Controller Id')]
+        [int]     $ControllerId = 65535,
+
+        [Parameter(
+            Mandatory = $false,
+            HelpMessage = 'NVMe/TCP IO Queue Number')]
+        [int]     $IoQueueNumber = 8,
+
+        [Parameter(
+            Mandatory = $false,
+            HelpMessage = 'NVMe/TCP IO Queue Size')]
+        [int]     $IoQueueSize = 256,
+
+
+        [Parameter(
+            Mandatory = $false,
+            HelpMessage = 'Keep Alive Timeout')]
+        [int]     $KeepAliveTimeout = 256,
+
+        [Parameter(
+            Mandatory = $false,
+            HelpMessage = 'Port Number')]
+        [int]     $PortNumber = 4420
+
+
+
+    )
+       
+    Write-Host "Connecting to targets via Storage Adapter on given ESXi host " $HostAddress;
+    Write-Host " " ;
+
+    $HostEsxcli = $null;
+ 
+    try {
+        $HostEsxcli = Get-EsxCli -VMHost $HostAddress 
+    }
+    catch {
+        throw "Failed to execute Get-EsxCli cmdlet on host $($HostAddress). Make sure valid ESXi IP/DNS is provided."
```

Comment:
When re-throwing  considering including the original error somewhere in the message `$_.Exception` to help with troubleshooting.

Reply:
Fixed, by adding the original exception to user facing message.